h2o-algos/src/test/java/hex/deeplearning/DeepLearningAutoEncoderCategoricalTest.java

```diff
@@ -84,16 +85,43 @@ public void run() {
 
     Assert.assertEquals(l2vec.mean(), mymodel.mse(), 1e-8*mymodel.mse());
 
+    // Create reconstruction
+    Log.info("Creating full reconstruction.");
+    final Frame recon_train = mymodel.score(train);
+    Assert.assertTrue(mymodel.testJavaScoring(train,recon_train,1e-5));
+
+    Frame df1 = mymodel.scoreDeepFeatures(train, 0);
+    Assert.assertTrue(df1.numCols() == 10);
```

Comment:
the frame will leak if the assertion fails

also I wouldn't use assertTrue for checking equality of numbers, assertEquals is better before it is more verbose

Reply:
Yes, that is true.